src/vector/feature.rs

```diff
@@ -177,6 +177,99 @@ impl<'a> Feature<'a> {
+            return Err(GdalError::InvalidFieldName {
+                field_name: field_name.to_string(),
+                method_name: "OGR_F_GetFieldIndex",
+            });
+        }
+
+        Ok(field_id)
+    }
+
+    /// Get the value of the specified field as a [`i32`].
+    ///
+    /// If the field is missing, returns [`GdalError::InvalidFieldName`].
+    ///
+    /// Returns `Ok(None)` if the field is null.
+    /// Returns `Ok(0)` on other kinds of errors.
+    ///
+    pub fn get_field_integer(&self, field_name: &str) -> Result<Option<i32>> {
+        let field_idx = self.field_idx_from_name(field_name)?;
+
+        if unsafe { gdal_sys::OGR_F_IsFieldNull(self.c_feature, field_idx) } != 0 {
+            return Ok(None);
+        }
+
+        let value = unsafe { gdal_sys::OGR_F_GetFieldAsInteger(self.c_feature, field_idx) };
+
+        Ok(Some(value))
+    }
+
+    /// Get the value of the specified field as a [`i64`].
+    ///
+    /// If the field is missing, returns [`GdalError::InvalidFieldName`].
+    ///
+    /// Returns `Ok(None)` if the field is null.
+    /// Returns `Ok(0)` on other kinds of errors.
+    ///
+    pub fn get_field_integer64(&self, field_name: &str) -> Result<Option<i64>> {
+        let field_idx = self.field_idx_from_name(field_name)?;
+
+        if unsafe { gdal_sys::OGR_F_IsFieldNull(self.c_feature, field_idx) } != 0 {
+            return Ok(None);
+        }
+
+        let value = unsafe { gdal_sys::OGR_F_GetFieldAsInteger64(self.c_feature, field_idx) };
+
+        Ok(Some(value))
+    }
+
+    /// Get the value of the specified field as a [`f64`].
+    ///
+    /// If the field is missing, returns [`GdalError::InvalidFieldName`].
+    ///
+    /// Returns `Ok(None)` if the field is null.
+    /// Returns `Ok(0.)` on other kinds of errors.
+    ///
+    pub fn get_field_double(&self, field_name: &str) -> Result<Option<f64>> {
+        let field_idx = self.field_idx_from_name(field_name)?;
+
+        if unsafe { gdal_sys::OGR_F_IsFieldNull(self.c_feature, field_idx) } != 0 {
+            return Ok(None);
+        }
+
+        let value = unsafe { gdal_sys::OGR_F_GetFieldAsDouble(self.c_feature, field_idx) };
+
+        Ok(Some(value))
+    }
+
+    /// Get the value of the specified field as a [`String`].
+    ///
+    /// If the field is missing, returns [`GdalError::InvalidFieldName`].
+    ///
+    /// Returns `Ok(None)` if the field is null.
+    ///
+    pub fn get_field_string(&self, field_name: &str) -> Result<Option<String>> {
+        let field_idx = self.field_idx_from_name(field_name)?;
+
+        if unsafe { gdal_sys::OGR_F_IsFieldNull(self.c_feature, field_idx) } != 0 {
+            return Ok(None);
+        }
+
+        let value = _string(unsafe { gdal_sys::OGR_F_GetFieldAsString(self.c_feature, field_idx) });
+
+        Ok(Some(value))
+    }
+
     #[cfg(feature = "datetime")]
     fn get_field_datetime(&self, field_id: c_int) -> Result<DateTime<FixedOffset>> {
```

Comment:
we could also rename this to field_as_datetime

Reply:
Well, this has a different signature as it takes the field index as a parameter. Moreover, it is not public.
Should I create a public function field_datetime with `field_name` as input?